util/git-diff.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+# Copyright 2020 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+import argparse
+from git import Repo
+
+
+def main():
+    repo = Repo(search_parent_directories=True)
+
+    parser = argparse.ArgumentParser(
+        description='Check for changes in repository')
+    parser.add_argument('--error-msg',
+                        default='::error Files differ.',
+                        required=False,
+                        help='custom exit code string')
+    args = parser.parse_args()
+
+    diff_to_head = repo.head.commit.diff(None)
+
+    # diff tree against working tree
+    for diff in diff_to_head:
+        print("::error file={}::Files differ ({})".format(
+            diff.b_path, diff.change_type))
```

Comment:
Can we pass the line number that the difference was found in like that comment ? 
https://github.com/golangci/golangci-lint-action/issues/119#issuecomment-738355857